src/plugins/visualizations/public/embeddable/visualize_embeddable_factory.tsx

```diff
@@ -118,9 +137,18 @@ export class VisualizeEmbeddableFactory
   public async create() {
     // TODO: This is a bit of a hack to preserve the original functionality. Ideally we will clean this up
     // to allow for in place creation of visualizations without having to navigate away to a new URL.
+    let originitingAppParam = this.currentAppId;
+    // TODO: Remove this after https://github.com/elastic/kibana/pull/63443
+    if (originitingAppParam === 'kibana') {
+      originitingAppParam += `:${window.location.hash.split(/[\/\?]/)[1]}`;
```

Comment:
Not a big fan of needing to know the exact positional order of the parameter in the URL: can you use one of the URL parsing libs we have, like `query-string` to convert this into a named argument?

Reply:
This whole section will thankfully not be necessary after @flash1293 is finished #63443. in the meantime, some URL parsing is necessary to grab `dashboard` out of: 
`#/dashboard/722b74f0-b882-11e8-a6d9-e546fe2bba5f ...`.

as far as I can see , it's not exactly possible to get a named argument for it using query-string, because it isn't really a param.  That said, something like this:

```
const urlSplit = window.location.href.split('/');
editUrl += urlSplit[urlSplit.indexOf('kibana#') + 1]
```

Can isolate 'dashboard' without the exact positional order, but I'm not sure if the change is worth it. Any thoughts?